stdlib/http/src/main/ballerina/http/service_endpoint.bal

```diff
@@ -164,13 +230,183 @@ public type KeepAlive "AUTO"|"ALWAYS"|"NEVER";
 @final public KeepAlive KEEPALIVE_NEVER = "NEVER";
 
 function Listener::init (ServiceEndpointConfiguration c) {
+    match c.authProviders {
+        AuthProvider[] providers => addAuthFiltersForSecureListener(c, self.instanceId);
+        () => {}
+    }
     self.config = c;
     var err = self.initEndpoint();
     if (err != null) {
         throw err;
     }
 }
 
+# Add authn and authz filters
+#
+# + config - `ServiceEndpointConfiguration` instance
+function addAuthFiltersForSecureListener(ServiceEndpointConfiguration config, string instanceId) {
+    // add authentication and authorization filters as the first two filters.
+    // if there are any other filters specified, those should be added after the authn and authz filters.
+    if (config.filters == null) {
+        // can add authn and authz filters directly
+        config.filters = createAuthFiltersForSecureListener(config, instanceId);
+    } else {
+        Filter[] newFilters = createAuthFiltersForSecureListener(config, instanceId);
+        // add existing filters next
+        int i = 0;
+        while (i < lengthof config.filters) {
+            newFilters[i + (lengthof newFilters)] = config.filters[i];
+            i = i + 1;
+        }
+        config.filters = newFilters;
+    }
+}
+
+# Create an array of auth and authz filters.
+#
+# + config - `ServiceEndpointConfiguration` instance
+# + return - Array of Filters comprising of authn and authz Filters
+function createAuthFiltersForSecureListener(ServiceEndpointConfiguration config, string instanceId) returns (Filter[]) {
```

Comment:
Shall we document the 'instanceId' parameter?

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/11173/commits/34216c7f1de2b3ee7fd63331d5384c928a3fa0a7